framework/packages/abstract-std/src/account.rs

```diff
@@ -359,6 +365,32 @@ pub struct ConfigResponse {
     pub module_factory_address: Addr,
 }
 
+#[cosmwasm_schema::cw_serde]
+pub struct ICS20PacketIdentifier {
+    pub channel_id: String,
+    pub sequence: u64,
+}
+
+impl<'a> PrimaryKey<'a> for ICS20PacketIdentifier {
+    /// channel id
+    type Prefix = String;
+
+    /// channel id
+    type SubPrefix = String;
+
+    /// sequence
+    type Suffix = u64;
+
+    // sequence
+    type SuperSuffix = u64;
+
+    fn key(&self) -> Vec<cw_storage_plus::Key> {
+        let mut keys = self.channel_id.key();
+        keys.extend(self.sequence.key());
+        keys
+    }
+}
```

Comment:
I remember you preferred to re-use default PrimaryKey implementations when possible, shouldn't we re-use  for `(T, U)` implementation here instead?

Reply:
yes. However, that doesn't work for values that are not tuples at the start, because we'd have to create the tuple to get the implementation. 
``` rust
error[E0515]: cannot return value referencing temporary value
   --> packages/abstract-std/src/account.rs:388:9
    |
388 |         (self.channel_id.clone(), self.sequence).key()
    |         ----------------------------------------^^^^^^
    |         |
    |         returns a value referencing data owned by the current function
    |         temporary value created here
    ```